### PR TITLE
Problem: new app options don't have affect

### PR DESCRIPTION
**Pull request by yihuang**
Solution:
- use DefaultBaseappOptions provided by sdk

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

**Review by mmsqe — APPROVED** [review]
oh, missed SetIAVLLazyLoading

File: cmd/cronosd/cmd/root.go [hunk 1]
```diff
@@ -220,61 +216,25 @@ type appCreator struct {
...
 // newApp is an AppCreator
 func (a appCreator) newApp(logger log.Logger, db dbm.DB, traceStore io.Writer, appOpts servertypes.AppOptions) servertypes.Application {
-	var cache sdk.MultiStorePersistentCache
-
-	if cast.ToBool(appOpts.Get(server.FlagInterBlockCache)) {
-		cache = store.NewCommitKVStoreCacheManager()
+	snapshotDir := filepath.Join(cast.ToString(appOpts.Get(flags.FlagHome)), "data", "snapshots")
+	if err := os.MkdirAll(snapshotDir, os.ModePerm); err != nil {
+		panic(err)
 	}
```
Comment (mmsqe) [inline cmd/cronosd/cmd/root.go:222]:
```suggestion
```

File: cmd/cronosd/cmd/root.go [hunk 2]
```diff
@@ -220,61 +216,25 @@ type appCreator struct {
 
 // newApp is an AppCreator
 func (a appCreator) newApp(logger log.Logger, db dbm.DB, traceStore io.Writer, appOpts servertypes.AppOptions) servertypes.Application {
-	var cache sdk.MultiStorePersistentCache
-
-	if cast.ToBool(appOpts.Get(server.FlagInterBlockCache)) {
-		cache = store.NewCommitKVStoreCacheManager()
+	snapshotDir := filepath.Join(cast.ToString(appOpts.Get(flags.FlagHome)), "data", "snapshots")
+	if err := os.MkdirAll(snapshotDir, os.ModePerm); err != nil {
+		panic(err)
```
Comment (mmsqe) [inline cmd/cronosd/cmd/root.go:221]: actually not sure why [sdk](https://github.com/cosmos/cosmos-sdk/blob/v0.46.9-rc.0/snapshots/store.go#L41) not panic if no dir

File: cmd/cronosd/cmd/root.go (hunk 1 above)
Comment (mmsqe) [inline cmd/cronosd/cmd/root.go:222]:
```suggestion
```

Comment (mmsqe) [reply cmd/cronosd/cmd/root.go:222]: sdk contains this [part](https://github.com/cosmos/cosmos-sdk/blob/v0.46.9-rc.0/server/util.go#L436)

Comment (yihuang) [reply cmd/cronosd/cmd/root.go:222]: no mkdir

Comment (mmsqe) [reply cmd/cronosd/cmd/root.go:222]: actually a bit different, [0o755](https://github.com/cosmos/cosmos-sdk/blob/v0.46.9-rc.0/snapshots/store.go#L39) vs ModePerm [0o777](https://pkg.go.dev/os)

Comment (yihuang) [reply cmd/cronosd/cmd/root.go:222]: yeah, we need to create the dir before open the db.

**Pull request merged**